### PR TITLE
Remove [direct] dependency on go.uber.org/atomic

### PR DESCRIPTION
**Pull request by zmb3**
Reintroduce the change from #16884 now that Go 1.19.2 has been released with the fix for 32-bit ARM.

This reverts the revert from  09de0ba5302c47763ff3970032e8f8da4251ef33.

### CODE REVIEW

**Review by GavinFrazar — APPROVED** [review]

**Review by jakule — APPROVED** [review]
Round 2 approved.

File: go.mod [hunk 1]
```diff
@@ -327,6 +326,7 @@ require (
 	go.opentelemetry.io/otel/exporters/otlp/internal/retry v1.10.0 // indirect
 	go.opentelemetry.io/otel/metric v0.32.1 // indirect
 	go.starlark.net v0.0.0-20200306205701-8dd3e2ee1dd5 // indirect
+	go.uber.org/atomic v1.7.0 // indirect
```
Comment (jakule) [inline go.mod:329]: @zmb3 Shouldn't you also increase the go version in the `go.mod` file? We explicitly want to use atomics from 1.19.

Comment (jakule) [reply go.mod:329]: I think that is also why CodeQL is complaining. UInt64 was introduced in Go 1.19 and the CodeQL build is using Go 1.18
https://pkg.go.dev/sync/atomic@master#Uint64

Comment (zmb3) [reply go.mod:329]: Good catch! Yep.

**Pull request merged**